crates/amaru-stores/src/rocksdb/mod.rs

```diff
@@ -337,23 +268,85 @@ fn with_prefix_iterator<
         }
         .map_err(|err| StoreError::Internal(err.into()))?;
     }
+    Ok(())
+}
 
-    db.commit().map_err(|err| StoreError::Internal(err.into()))
+struct RocksDBTransactionalContext<'a> {
+    db: &'a RocksDB,
+    transaction: Transaction<'a, OptimisticTransactionDB>,
 }
 
-impl Store for RocksDB {
-    fn for_epoch(&self, epoch: Epoch) -> Result<impl Snapshot, StoreError> {
-        Self::for_epoch_with(self.dir.as_path(), epoch)
+impl TransactionalContext<'_> for RocksDBTransactionalContext<'_> {
+    fn commit(self) -> Result<(), StoreError> {
+        let res = self
+            .transaction
+            .commit()
+            .map_err(|err| StoreError::Internal(err.into()));
+        self.db.transaction_committed();
+        res
     }
 
-    fn tip(&self) -> Result<Point, StoreError> {
-        self.db
-            .get(KEY_TIP)
-            .map_err(|err| StoreError::Internal(err.into()))?
-            .map(|bytes| cbor::decode(&bytes))
-            .transpose()
-            .map_err(|err| StoreError::Tip(TipErrorKind::Undecodable(err)))?
-            .ok_or(StoreError::Tip(TipErrorKind::Missing))
+    #[instrument(
+        level = Level::TRACE,
+        skip_all,
+    )]
+    fn reset_fees(&self) -> Result<(), StoreError> {
+        self.with_pots(|mut row| {
```

Comment:
Those 4 fns should be default fns on the `Transaction` trait. Failed so far dur to intricacies with `iter_borrow` .

Reply:
I would perhaps more challenge the idea of them being in the store to begin with? I rather keep the store interface minimal, and have those defined as _helpers_ on top.

Reply:
So just regular functions taking a `Transaction` as parameter? And not default trait method? Makes sense.